### PR TITLE
Update references to GitHub Enterprise Server in UI and code comments

### LINKED ISSUE #16952: Update references to GitHub Enterprise Server in GitHub Desktop UI and code comments
### The problem

We can help users and developers build a mental model of GitHub's products and also help them succeed as they onboard with Desktop by updating language in the UI and code to reflect GitHub Enterprise Server's current name.

UI text and code comments for GitHub Desktop reference "[GitHub Enterprise](https://github.com/search?q=repo%3Adesktop%2Fdesktop%20%22GitHub%20Enterprise%22&type=code)" instead of "[GitHub Enterprise **Server**](https://docs.github.com/enterprise-server/admin/overview/about-github-enterprise-server)". The product's name used to be just "GitHub Enterprise", but "GitHub Enterprise" now describes [this plan](https://docs.github.com/get-started/learning-about-github/githubs-products#github-enterprise), which comprises both GitHub Enterprise Cloud _and_ GitHub Enterprise Server.

There's at least one exception to this: [here](https://docs.github.com/assets/cb-46309/mw-1440/images/help/desktop/sign-in-ghes.webp), during [initial authentication](https://docs.github.com/desktop/installing-and-configuring-github-desktop/installing-and-authenticating-to-github-desktop/authenticating-to-github-in-github-desktop#authenticating-an-account-on-github-enterprise-server), we'd want to refer to just "GitHub Enterprise" since we're describing both GitHub Enterprise Server and GitHub AE.

### Release version

All

### Operating system

All

### Steps to reproduce the behavior

There are [many locations in code](https://github.com/search?q=repo%3Adesktop%2Fdesktop%20%22GitHub%20Enterprise%22&type=code) where we refer to "GitHub Enterprise". One of the first flows where someone would see this is [authentication](https://docs.github.com/en/desktop/installing-and-configuring-github-desktop/installing-and-authenticating-to-github-desktop/authenticating-to-github-in-github-desktop#authenticating-an-account-on-github-enterprise-server).

### PR DESCRIPTION
**Pull request by mattpollard**
## Description

Updates outdated references to "GitHub Enterprise" to "GitHub Enterprise _Server_". 

- Closes https://github.com/desktop/desktop/issues/16952.
- Closes https://github.com/github/desktop/issues/711.

## Release notes

Notes: To reflect the current name of the product and not the broader plan that includes GitHub Enterprise Cloud on GitHub.com, some strings that referenced "GitHub Enterprise" now reference "GitHub Enterprise Server".

### CODE REVIEW

**Comment by niik** [comment]
Bah, hit the wrong button. I meant to approve the workflows but instead I made the draft ready to review. Sorry about that!

**Comment by HICUELA** [comment]
IMG_0014
IMG_0015

**Comment by niik** [comment]
@mattpollard Is there anything else you'd like to add to this PR? Looks like there's some linting issue but we can get them sorted for you if you'd like

File: app/src/lib/stats/stats-database.ts [hunk 1]
```diff
@@ -77,10 +77,10 @@ export interface IDailyMeasures {
   /** The number of times the user pushes with `--force-with-lease` to GitHub.com */
   readonly dotcomForcePushCount: number
 
-  /** The number of times the user pushed to a GitHub Enterprise instance */
+  /** The number of times the user pushed to a GitHub Enterprise server instance */
```
Comment (mattpollard) [inline app/src/lib/stats/stats-database.ts]:
```suggestion
  /** The number of times the user pushed to a GitHub Enterprise Server instance */
```

**Comment by mattpollard** [comment]
Looks like there are a few more references I need to update. I'd appreciate help with CI, though! 😄

**Comment by niik** [comment]
> I'd appreciate help with CI, though!

Happy to help, I've fixed the linting issues

**Comment by mattpollard** [comment]
@niik 👋🏻 You're welcome to kick off the workflow runs! After those are done, if everything looks good, I'll promote this from a draft and it'll be ready for review ☺️

**Comment by Tantaia111** [comment]
Cap nhap

**Comment by tidy-dev** [comment]
@mattpollard I kicked off the checks and just addressed the lint errors. Is this good to go?

**Comment by mattpollard** [comment]
Sorry, @tidy-dev... I was out for a few weeks! I've updated a few more references, so this should be ready for another workflow run. Hopefully, after that, we can get this reviewed 😄

**Comment by niik** [comment]
@mattpollard Looks like all checks passed. Feel free to flip the ready for review switch whenever you'd like and we'll take a look!

**Review by sergiou87 — COMMENTED** [review]
I'm still lost with so many Enterprise products 😂  But does this make sense?

2023-08-22 at 18 08

The title says GHES but the description also mentions GHAE.

**Comment by mattpollard** [comment]
~Yep! The text you've highlighted is still sensible.~

Oof, I just noticed that the bold text atop that notice should not be "GitHub Enterprise Server". Lemme fix that...

**Comment by mattpollard** [comment]
@sergiou87, I pushed https://github.com/desktop/desktop/pull/16953/commits/7ab5195014a25a01e6980cbe95c83e9016b772fd to address this. However, it's a long string 😅 Could you see how it renders?

We _could_ go with just "GitHub Enterprise", but I think titling the second heading just "GitHub Enterprise" doesn't address some of the internal feedback we've gotten that the sign-in options are confusing.

**Comment by sergiou87** [comment]
This is what it looks like:

image


But makes me wonder… does this kind of distinction between GHES and GHAE also apply in the rest of strings you changed in this PR? 🤔

**Comment by mattpollard** [comment]
That's a good call. I'll look back through the changes in the next week or so and determine which really refer to GHES _instances_, versus GitHub Enterprise _the plan_ 👀

**Comment by xhibika-ann** [comment]
I am trying to understand. Really I am. Thanks for all you do.

**Comment by mattpollard** [comment]
We need to think a little more about this change. We're tracking that internally. I'll close this PR 🙏🏻 Thanks for your help, everyone!

**Pull request closed without merging**